### ISSUE TITLE
[code question] About pts_out_init.detach()

### ISSUE THREAD

**Issue by JanLin0817** [microsoft/RepPoints#11]
Hello, what is the detach usage in class RepPointsHead
https://github.com/microsoft/RepPoints/blob/cac17da18b54c800db1565cf39a2723f6755b96c/src/reppoints_head/reppoints_head.py#L276

Is this try to prevent `pts_out_init` backward propagation multiple times?

**Comment by muditchaudhary** [comment]
As far as I can understand, it is to balance the loss terms for localization and classification.

Check https://github.com/microsoft/RepPoints/issues/6 for the information.
